printrun/projectlayer.py

```diff
@@ -255,213 +259,246 @@ def fit(ev):
         self.interval = wx.TextCtrl(settingsGroup, -1, str(self._get_setting("project_interval", "0.5")), size = (125, -1))
         self.interval.Bind(wx.EVT_TEXT, self.update_interval)
-        self.interval.SetToolTip("How long each slice should be displayed.")
+        self.interval.SetToolTip(_("How long each slice should be displayed."))
         fieldsizer.Add(self.interval, pos = (1, 1))
 
-        fieldsizer.Add(wx.StaticText(settingsGroup, -1, "Blank (s):"), pos = (2, 0), flag = wx.ALIGN_CENTER_VERTICAL)
+        fieldsizer.Add(wx.StaticText(settingsGroup, -1, _("Blank (s):")), pos = (2, 0), flag = wx.ALIGN_CENTER_VERTICAL | wx.ALIGN_RIGHT)
         self.pause = wx.TextCtrl(settingsGroup, -1, str(self._get_setting("project_pause", "0.5")), size = (125, -1))
         self.pause.Bind(wx.EVT_TEXT, self.update_pause)
-        self.pause.SetToolTip("The pause length between slices. This should take into account any movement of the Z axis, plus time to prepare the resin surface (sliding, tilting, sweeping, etc).")
+        self.pause.SetToolTip(_("The pause length between slices. This should take into account any movement of the Z axis, plus time to prepare the resin surface (sliding, tilting, sweeping, etc)."))
         fieldsizer.Add(self.pause, pos = (2, 1))
 
-        fieldsizer.Add(wx.StaticText(settingsGroup, -1, "Scale:"), pos = (3, 0), flag = wx.ALIGN_CENTER_VERTICAL)
+        fieldsizer.Add(wx.StaticText(settingsGroup, -1, _("Scale:")), pos = (3, 0), flag = wx.ALIGN_CENTER_VERTICAL | wx.ALIGN_RIGHT)
         self.scale = wx.SpinCtrlDouble(settingsGroup, -1, initial = self._get_setting('project_scale', 1.0), inc = 0.1, size = (125, -1))
         self.scale.SetDigits(3)
         self.scale.Bind(wx.EVT_SPINCTRLDOUBLE, self.update_scale)
-        self.scale.SetToolTip("The additional scaling of each slice.")
+        self.scale.SetToolTip(_("The additional scaling of each slice."))
         fieldsizer.Add(self.scale, pos = (3, 1))
 
-        fieldsizer.Add(wx.StaticText(settingsGroup, -1, "Direction:"), pos = (4, 0), flag = wx.ALIGN_CENTER_VERTICAL)
-        self.direction = wx.ComboBox(settingsGroup, -1, choices = ["Top Down", "Bottom Up"], value = self._get_setting('project_direction', "Top Down"), size = (125, -1))
-        self.direction.Bind(wx.EVT_COMBOBOX, self.update_direction)
-        self.direction.SetToolTip("The direction the Z axis should move. Top Down is where the projector is above the model, Bottom up is where the projector is below the model.")
+        fieldsizer.Add(wx.StaticText(settingsGroup, -1, _("Direction:")), pos = (4, 0), flag = wx.ALIGN_CENTER_VERTICAL | wx.ALIGN_RIGHT)
+        self.direction = wx.Choice(settingsGroup, -1, choices = [_('Top Down'), _('Bottom Up')], size = (125, -1))
+        saved_direction = self._get_setting('project_direction', 0)
+        try:
+            int(saved_direction) # This setting used to be a string, so older settings need to be overwritten
+        except:    
```

Comment:
ad 1. I stumbled across this when adding all the _() translation markups. Since the string of "Top Down" etc. was used in if statements, translating it probably wouldn't work well. So I thought the cleaner solution would be to replace the wx.ComboBox with a wx.Choice and work with the choice indexes. (and from a ux point of view I always prefer wx.Choice if it were possible)
ad 2. Yep, that is supposed to be there. I completely forgot.
ad 3. Hmm, 'Top Down' was the fallback option before. In my opinion, it is not a critical setting, and I went with the uncomplicated solution. But now that you mention it, I will just change it based on the string. Good point!

Reply:
Thanks for the explanations. Good catch. Changes agreed then.